### ISSUE TITLE
Evaluate support of Pydantic-v2 generated JSON Schemas

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-web#479]
When testing the schemas in https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/737, we should be aware of how Pydantic V2 changes handling of `Optional` attributes (which includes the use of `anyOf`, and the fact that it seems that attributes are always included). The new behavior is sometimes confusing, see:
* https://github.com/pydantic/pydantic/issues/7161
* https://github.com/pydantic/pydantic/issues/8394

It's likely that this produces issues in the form generator.
By taking the example in https://github.com/pydantic/pydantic/issues/7161#issue-1854576770, we end up with this JSON Schema:
```json
{
  "properties": {
    "field": {
      "anyOf": [
        {
          "type": "string"
        },
        {
          "type": "null"
        }
      ],
      "default": null,
      "title": "Field"
    }
  },
  "title": "Model",
  "type": "object"
}
```

If we try it out in https://rjsf-team.github.io/react-jsonschema-form/, we find this behavior:

![image](https://github.com/fractal-analytics-platform/fractal-web/assets/3862206/f941a2b9-bbcc-4276-b393-dce92c33a47b)
![image](https://github.com/fractal-analytics-platform/fractal-web/assets/3862206/ca8f7fa7-1c23-4f66-8cc2-d3fc5249e206)

which IMO is a suboptimal UX, because it forces a two-steps process: first I need to decide whether to include the argument, and then I can decide whether/how to fill it.

---

TLDR We'll need to keep an eye on this, when experimenting with schema-based form generators.

**Comment by tcompa** [comment]
This issue was a FYI, but the plan is rather to make sure that fractal-tasks-core produces appropriate JSON Schemas (that is, without the additional `anyOf` pattern for nullable field) even with Pydantic v2.

Closing.
